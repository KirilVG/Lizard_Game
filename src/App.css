.App {
  text-align: center;
}

.App-logo {
  height: 40vmin;
  pointer-events: none;
}

@media (prefers-reduced-motion: no-preference) {
  .App-logo {
    animation: App-logo-spin infinite 20s linear;
  }
}

.App-header {
  background-color: #282c34;
  min-height: 100vh;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  font-size: calc(10px + 2vmin);
  color: white;
}

.App-link {
  color: #61dafb;
}

@keyframes App-logo-spin {
  from {
    transform: rotate(0deg);
  }
  to {
    transform: rotate(360deg);
  }
}

input {
  align-self: center;
  padding: 12px 20px;
  margin: 8px 0;
  box-sizing: border-box;
  border: none;
  border-bottom: 2px solid black;
}

.content{
  border: 1px solid gray;
  margin-top:180px;
  margin-left:380px;
  background-color: #464749;
  width:333px;
  height: 352px;
  padding: 58px 76px 0 76px;
  font: 12px Arial, Helvetica, sans-serif;
}

.nameField{
  display:inline-block;
  width:80px;
  text-align: right;
  padding: 14px 10px 0 0;
  margin:0 0 7px 0;
}
.inputField{
  display:inline-block;
  width:130px;
  margin:0;
  margin:0 0 7px 0;
}
.buttonStart{
  background-color: black;
  border: none;
  color: white;
  padding: 15px 32px;
  text-align: center;
  text-decoration: none;
  display: inline-block;
  font-size: 16px;
  border-radius: 5%;
}

.playButton{
  height: 7vh;
  width: 20vh;
  font-Size: 4vh;
}

.glassCard{
  backdrop-filter: blur(15px);
  background-Color: rgba(255,255,255,.2) !important;
  border-radius: 3em !important;
  border: 2px solid rgba(255,255,255,.1);
  background-clip: padding-box;
  box-shadow: 10px 10px 10px rgba(46, 54, 68, 0.03);
}
