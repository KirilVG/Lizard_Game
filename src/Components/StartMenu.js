import React from "react";
import * as myConstants from "./Constants";

function StartMenu(props) {
  const [inputText, setInputText] = React.useState("3");
  const [inputUsername, setInputUsername] = React.useState("Noname");

  const handleInput = (event) => {
    setInputText(event.target.value);
  };

  const handleUsernameInput = (event) => {
    setInputUsername(event.target.value);
  };

  const clickStart = () => {
    let lanesNum = Number(inputText);
    if (!lanesNum) {
      alert("lanes should be a number");
    } else if (lanesNum % 1 !== 0) {
      alert("lanes should be a whole number");
    } else if (lanesNum < myConstants.minLanesNum) {
      alert(`lanes should not be less than ${myConstants.minLanesNum}`);
    } else if (lanesNum > myConstants.maxLanesNum) {
      alert(`lanes should not be more than ${myConstants.maxLanesNum}`);
    } else {
      props.onClick({lanesNum: lanesNum, username:inputUsername});
    }
  };

  return (
    <div>
      <h1>Welcome to Run Kiro the Lizard.</h1>
      <label>
        Lanes:
        <input onChange={handleInput} placeholder="3" />
      </label>
      <label>
        Username:
        <input onChange={handleUsernameInput} placeholder="Noname" />
      </label>
      <button onClick={clickStart}>Start</button>
    </div>
  );
}

export default StartMenu;
