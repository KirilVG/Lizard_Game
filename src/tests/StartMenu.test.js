import StartMenu from "../Components/StartMenu";
import App from "../App";
import { fireEvent, render, screen } from "@testing-library/react";
import renderer from "react-test-renderer";

test("start menu renders title element", () => {
  render(<App />);
  const linkElement = screen.getByText(/Welcome to run Kiro the lizard!/i);

  expect(linkElement).toBeInTheDocument();
});

test("start menu renders lanes input", () => {
  render(<App />);
  const lanesInput = screen.getByLabelText(/Number of lanes/i);

  expect(lanesInput).toBeInTheDocument();
});

test("start menu renders username input", () => {
  render(<App />);
  const usernameInput = screen.getByLabelText(/Username/i);

  expect(usernameInput).toBeInTheDocument();
});

test("start menu renders button", () => {
  render(<App />);
  const buttonEl = screen.getByRole("button");

  expect(buttonEl).toBeInTheDocument();
});

test("lanes input should change", () => {
  render(<App />);
  const testValue = "a";
  const lanesInput = screen.getByLabelText(/Number of lanes/i);

  fireEvent.change(lanesInput, { target: { value: testValue } });

  expect(lanesInput.value).toBe(testValue);
});

test("lanes input should set error prop to true when play button is pressed and Lanes input does not contain a number", () => {
  render(<App />);
  const lanesInput = screen.getByLabelText(/Number of lanes/i);
  const testValue = "a";
  const buttonEl = screen.getByRole("button");

  fireEvent.change(lanesInput, { target: { value: testValue } });
  fireEvent.click(buttonEl);

  const linkElement = screen.getByText(/lanes should be a number/i);

  expect(linkElement).toBeInTheDocument();
});

test("lanes input should set error prop to true when play button is pressed and Lanes input contains a number less than the minimum", () => {
  render(<App />);
  const lanesInput = screen.getByLabelText(/Number of lanes/i);
  const testValue = "2";
  const buttonEl = screen.getByRole("button");

  fireEvent.change(lanesInput, { target: { value: testValue } });
  fireEvent.click(buttonEl);

  const linkElement = screen.getByText(/lanes should not be less than 3/i);

  expect(linkElement).toBeInTheDocument();
});

test("lanes input should set error prop to true when play button is pressed and Lanes input contains a number more than the maximum", () => {
  render(<App />);
  const lanesInput = screen.getByLabelText(/Number of lanes/i);
  const testValue = "102";
  const buttonEl = screen.getByRole("button");

  fireEvent.change(lanesInput, { target: { value: testValue } });
  fireEvent.click(buttonEl);

  const linkElement = screen.getByText(/lanes should not be more than 100/i);

  expect(linkElement).toBeInTheDocument();
});

test("username input should change", () => {
  render(<App />);
  const usernameInput = screen.getByLabelText(/Username/i);
  const testValue = "User";

  fireEvent.change(usernameInput, { target: { value: testValue } });

  expect(usernameInput.value).toBe(testValue);
});

it("start menu renders correctly", () => {
  const lanesNum = null;
  const username = null;
  const onClick = () => {};
  const tree = renderer
    .create(
      <StartMenu onClick={onClick} lanesNum={lanesNum} username={username} />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
