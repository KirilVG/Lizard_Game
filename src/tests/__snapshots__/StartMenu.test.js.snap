// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`start menu renders correctly 1`] = `
<div
  className="d-flex justify-content-center align-items-center backgroundIMG"
  style={
    Object {
      "backgroundImage": "url(\\"https://live.staticflickr.com/65535/48701594348_e099758f0c_b.jpg\\")",
    }
  }
>
  <div
    className="mask darkMask d-flex justify-content-center align-items-center h-100"
  >
    <div
      className="container row justify-content-center"
    >
      <div
        className="col-12 col-md-10 col-lg-7 col-xl-6"
      >
        <div
          className="MuiFormControl-root card glassCard card-body inputForm css-1nrlq1o-MuiFormControl-root"
        >
          <h2
            className="text-center mb-5 formItem"
          >
            Welcome to run Kiro the lizard!
          </h2>
          <div
            className="formStack css-pnq1z9-MuiStack-root"
          >
            <div
              className="MuiFormControl-root MuiTextField-root css-19lnug-MuiFormControl-root-MuiTextField-root"
            >
              <label
                className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
                data-shrink={false}
                htmlFor="outlined-basic"
                id="outlined-basic-label"
              >
                Number of lanes
              </label>
              <div
                className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
                onClick={[Function]}
              >
                <input
                  aria-invalid={false}
                  autoFocus={false}
                  className="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
                  defaultValue={null}
                  disabled={false}
                  id="outlined-basic"
                  onAnimationStart={[Function]}
                  onBlur={[Function]}
                  onChange={[Function]}
                  onFocus={[Function]}
                  required={false}
                  type="text"
                />
                <fieldset
                  aria-hidden={true}
                  className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                >
                  <legend
                    className="css-yjsfm1"
                  >
                    <span>
                      Number of lanes
                    </span>
                  </legend>
                </fieldset>
              </div>
            </div>
            <div
              className="MuiFormControl-root MuiTextField-root css-19lnug-MuiFormControl-root-MuiTextField-root"
            >
              <label
                className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
                data-shrink={false}
                htmlFor="outlined-basic"
                id="outlined-basic-label"
              >
                Username
              </label>
              <div
                className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
                onClick={[Function]}
              >
                <input
                  aria-invalid={false}
                  autoFocus={false}
                  className="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
                  defaultValue={null}
                  disabled={false}
                  id="outlined-basic"
                  onAnimationStart={[Function]}
                  onBlur={[Function]}
                  onChange={[Function]}
                  onFocus={[Function]}
                  required={false}
                  type="text"
                />
                <fieldset
                  aria-hidden={true}
                  className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                >
                  <legend
                    className="css-yjsfm1"
                  >
                    <span>
                      Username
                    </span>
                  </legend>
                </fieldset>
              </div>
            </div>
            <button
              className="btn btn-light playButton"
              onClick={[Function]}
              type="button"
            >
              Play
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
